sdk/eventhub/Azure.Messaging.EventHubs/tests/Snippets/Sample10_MockingReadingEventsTests.cs

```diff
@@ -0,0 +1,254 @@
+        }
+
+        /// <summary>
+        ///   Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ReadPartitionWithReceiver()
+        {
+            await using var scope = await EventHubScope.CreateAsync(1);
+
+            #region Snippet:EventHubs_Sample10_MockingReadPartitionWithReceiver
+
+            var mockEventHubProducerClient = new Mock<EventHubProducerClient>();
+            mockEventHubProducerClient
+                .Setup(c => c.GetPartitionIdsAsync(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(new string[1] { "sample id" });
+
+            var mockEventHubPartitionReceiver = new Mock<PartitionReceiver>();
+            var mockBatchData = GenerateBatchData();
+            mockEventHubPartitionReceiver
+                .Setup(r => r.ReceiveBatchAsync(It.IsAny<int>(), It.IsAny<TimeSpan>(), It.IsAny<CancellationToken>()))
+                .ReturnsAsync(mockBatchData)
+                .Verifiable();
+
+            using CancellationTokenSource cancellationSource = new CancellationTokenSource();
+            cancellationSource.CancelAfter(TimeSpan.FromSeconds(30));
+
+            string firstPartition;
+
+            await using (var producer = mockEventHubProducerClient.Object)
+            {
+                firstPartition = (await producer.GetPartitionIdsAsync()).First();
+            }
+
+            var receiver = mockEventHubPartitionReceiver.Object;
+            try
+            {
+                while (!cancellationSource.IsCancellationRequested)
+                {
+                    int batchSize = 50;
+                    TimeSpan waitTime = TimeSpan.FromSeconds(1);
+
+                    IEnumerable<EventData> eventBatch = await receiver.ReceiveBatchAsync(
+                        batchSize,
+                        waitTime,
+                        cancellationSource.Token);
+
+                    //do we need assertion like this?
+                    for (var index = 0; index < mockBatchData.Count(); ++index)
+                    {
+                        Assert.AreEqual(mockBatchData.ElementAt(index).EventBody.ToString(), eventBatch.ElementAt(index).EventBody.ToString());
+                    }
+                }
+            }
+            catch (TaskCanceledException)
+            {
+                // This is expected if the cancellation token is
+                // signaled.
+            }
+            finally
+            {
+                await receiver.CloseAsync();
+            }
+
+            #endregion
+        }
+
+        /// <summary>
+        ///   Performs basic unit test validation of the contained snippet.
+        /// </summary>
+        ///
+        [Test]
+        public async Task ReadPartitionTrackLastEnqueued()
+        {
+            await using var scope = await EventHubScope.CreateAsync(1);
+
+            #region Snippet:EventHubs_Sample10_MockingReadPartitionTrackLastEnqueued
+
```

Comment:
@jsquire ,
In the lines 149 -180, I've demonstrated how to use EventHubsModelFactory to create Event Hub meta data types and also mocked them.
Is this okay?
Or should we look at breaking them down to individuals?